### PR TITLE
DAOS-2512 Added snapshot harasser

### PR DESCRIPTION
**Pull request by mjean308**
Test-tag-hw:pr,hw soak_harasser

Signed-off-by: Maureen Jean <maureen.jean@intel.com>

### CODE REVIEW

File: src/tests/ftest/soak/soak.yaml [hunk 1]
```diff
@@ -46,154 +77,159 @@ container_reserved:
...
+        - "SX"
+        # - "RP_2GX"
+        # - "RP_3GX"
+        # - "RP_4GX"
+fio_stress:
+  global:
+    name: 'global'
+    directory: "/tmp/daos_dfuse"
+    ioengine: 'libaio'
+    thread: 16
```
Comment (saurabhtandan) [inline src/tests/ftest/soak/soak.yaml]: needs to be 1. It is a bool

Comment (mjean308) [reply src/tests/ftest/soak/soak.yaml]: Saurabh, I have made the changes in PR-1601 and then I can merge if these ever get submitted

File: src/tests/ftest/soak/soak.py [hunk 2]
```diff
@@ -517,18 +719,27 @@ def run_soak(self, test_param):
...
         self.task_list = self.params.get("taskspernode", test_param + "*")
-
+        self.h_list = self.params.get("harasserlist", test_param + "*")
         job_list = self.params.get("joblist", test_param + "*")
         pool_list = self.params.get("poollist", test_param + "*")
         rank = self.params.get("rank", "/run/container_reserved/*")
-        obj_class = self.params.get(
-            "object_class", "/run/container_reserved/*")
+
+        if self.h_list is not None and "rebuild" in self.h_list:
```
Comment (amjustin13) [inline src/tests/ftest/soak/soak.py]: This line looks like it's doing what is being done in the `is_harasser()` method. Here you could do:
`if self.is_harasser("rebuild"):` similar to how you are doing it in line 341.

Comment (mjean308) [reply src/tests/ftest/soak/soak.py]: yes... make sense...  I have a function I should use it :)

File: src/tests/ftest/soak/soak.py [hunk 3]
```diff
@@ -125,6 +135,186 @@ def get_remote_logs(self):
...
+        Args:
+            harasser (list): list of harassers to launch
+            pools (TestPool): pool obj
+
+        """
+        job = None
+        # Launch harasser after one complete pass
+        for harasser in harassers:
+            if harasser == "rebuild":
+                method = self.launch_rebuild
```
[the remark this answers is not in the document]
Comment (mjean308) [reply src/tests/ftest/soak/soak.py]: The method is building up the parameters to pass to Thread.
 job = threading.Thread(
                target=method, args=param_list, name=name)
method = launch_rebuild
args = ranks, pools

So I think this is ok as is

File: src/tests/ftest/soak/soak.py (hunk 2 above)
Comment (mjean308) [reply src/tests/ftest/soak/soak.py]: Please review changes  thanks!

File: src/tests/ftest/soak/soak.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (amjustin13) [reply src/tests/ftest/soak/soak.py]: Ah. I see, didn't catch that.

**Pull request merged**